### PR TITLE
Upgrade torchmetrics to 0.11.1. Add ROC metrics for category features. Add sequence accuracy, char error rate, and perplexity metrics for text features.

### PR DESCRIPTION
**Pull request by justinxzhao**
Additional changes:
- Move prediction tensor input name (for metrics) to a separate registry, set at metric registration time.
- Move metric objective (min or max) to a separate registry, set at metric registration time.

Sample of new metrics.
```
╒════════════╤═══════════════════╤════════╤══════════════╤═════════════════════╤══════════════════╕
│ Survived   │   char_error_rate │   loss │   perplexity │   sequence_accuracy │   token_accuracy │
╞════════════╪═══════════════════╪════════╪══════════════╪═════════════════════╪══════════════════╡
│ train      │            0.0324 │ 0.0799 │       3.3581 │              0.8391 │           0.9678 │
├────────────┼───────────────────┼────────┼──────────────┼─────────────────────┼──────────────────┤
│ validation │            0.0449 │ 0.0954 │       3.3790 │              0.7771 │           0.9554 │
├────────────┼───────────────────┼────────┼──────────────┼─────────────────────┼──────────────────┤
│ test       │            0.0371 │ 0.0835 │       3.3656 │              0.8200 │           0.9640 │
╘════════════╧═══════════════════╧════════╧══════════════╧═════════════════════╧══════════════════╛
```

### CODE REVIEW

**Review by jppgks — APPROVED** [review]
Thanks for this refactor, Justin!

File: requirements.txt [hunk 1]
```diff
@@ -24,7 +24,7 @@ marshmallow
 marshmallow-jsonschema
 marshmallow-dataclass==8.5.5
 tensorboard
-torchmetrics<0.9
+torchmetrics
```
Comment (tgaddair) [inline requirements.txt:27]: Set lower bound on `torchmetrics>=0.11`

**Review by tgaddair — APPROVED** [review]

**Review by arnavgarg1 — APPROVED** [review]
Nice clean up!

**Comment by tgaddair** [comment]
@justinxzhao any blockers on merging this?

File: requirements.txt (hunk 1 above)
Comment (justinxzhao) [reply requirements.txt]: Done.

**Comment by justinxzhao** [comment]
> @justinxzhao any blockers on merging this?

No blockers. Just rebased and can merge it after tests pass.

**Pull request merged**